test/integration/finance/api/payment_callbacks/stripe_callbacks_controller_test.rb

```diff
@@ -61,11 +61,11 @@ class CreateTest < self
     end
 
     test 'cannot find payment intent' do
-      stripe_event = self.stripe_event(type: 'payment_intent.requires_action', payment_intent_data: { id: 'non-existent-payment-intent-id' })
+      stripe_event = self.stripe_event(type: 'payment_intent.succeeded', payment_intent_data: { id: 'non-existent-payment-intent-id' })
```

Comment:
I think this test was wrong, it was supposed to check the controller returned a 404 when the payment is not found, but instead of that it was sending a `requires_action` event, which is not allowed. The test passed because not allowed events also return a 404, but the error was being returned before even checking whether the payment existed or not. Now it really checks the payment and returns a 204 when it doesn't exits.

Reply:
I agree. Also, [this test](https://github.com/3scale/porta/blob/master/test/integration/finance/api/payment_callbacks/stripe_callbacks_controller_test.rb#L55-L61) is checking already what happens when an invalid event is received already.